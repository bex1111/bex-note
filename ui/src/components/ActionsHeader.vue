<script setup>

import PrimeButton from "@/main";
</script>

<template>
  <h3>test</h3>
 <prime-button> test </prime-button>
</template>

<style scoped>

</style>