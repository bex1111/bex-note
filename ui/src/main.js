import { createApp } from 'vue'
import App from './App.vue'
import PrimeVue from 'primevue/config';
import Button from 'primevue/button';
import Select from 'primevue/select';
import Toolbar from 'primevue/toolbar';
import Card from 'primevue/card';
import InputText from 'primevue/inputtext';
import FloatLabel from 'primevue/floatlabel';
import Fluid from 'primevue/fluid';
import Dialog from 'primevue/dialog';
import Aura from '@primeuix/themes/aura';
import 'primeicons/primeicons.css'
import { createPinia } from 'pinia'
import {useAuthorizationStore} from "./store/authorization";
import './style.css'

const app=createApp(App)

const pinia = createPinia()
app.use(pinia)
export const tokenStore = useAuthorizationStore()

app.use(PrimeVue,{
    theme: {
        preset: Aura
    }
});
app.component('prime-button', Button);
app.component('prime-select', Select);
app.component('prime-toolbar', Toolbar);
app.component('prime-card', Card);
app.component('prime-input', InputText);
app.component('prime-float-label', FloatLabel);
app.component('prime-fluid', Fluid);
app.component('prime-dialog', Dialog);

app.mount('#app')
