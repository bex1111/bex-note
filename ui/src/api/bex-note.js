import axios from 'axios';
import {tokenStore} from '../main';


export const getNoteList = async () => {
    const response = await axios.get('http://localhost:1080/api/internal/note/list', {
        headers: {
            'x-auth-token': tokenStore.token
        }
    });
    return response.data;
}

export const authorize = async (username, password) => {
    const response = await axios.post('http://localhost:1080/api/authorization', {
        username,
        password
    });
    return response.data;
}


export const deleteNote = async (title) => {
    const response = await axios.delete('http://localhost:1080/api/internal/note/delete', {
        headers: {
            'x-auth-token': tokenStore.token
        },
        data: { title }
    });
    return response.data;
}

export const saveNote = async (title,content) => {
    const response = await axios.delete('http://localhost:1080/api/internal/note/save', {
        headers: {
            'x-auth-token': tokenStore.token
        },
        data: { title, content }
    });
    return response.data;
}


