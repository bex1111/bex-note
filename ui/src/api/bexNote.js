import axios from 'axios';
import {tokenStore} from '../main';


export const getNoteList = async () => {
    const response = await axios.get('/api/internal/note/list', {
        headers: {
            'x-auth-token': tokenStore.token
        }
    });
    return response.data;
}

export const authorize = async (username, password) => {
    const response = await axios.post('/api/authorize', {
        username,
        password
    });
    return response.data;
}


export const deleteNote = async (title) => {
    const response = await axios.delete('/api/internal/note/delete', {
        headers: {
            'x-auth-token': tokenStore.token
        },
        data: {title}
    });
    return response.data;
}

export const saveNote = async (title, content) => {
    const response = await axios.post('/api/internal/note/save', {
        data: {title, content}
    }, {
        headers: {
            'x-auth-token': tokenStore.token
        }
    });
    return response.data;
}

export const getContent = async (title) => {
    const response = await axios.post('/api/internal/note/content', {
            data: {title}
        },
        {
            headers: {
                'x-auth-token': tokenStore.token
            }
        });
    return response.data;
}
