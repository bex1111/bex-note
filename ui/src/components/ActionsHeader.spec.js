import {mount} from '@vue/test-utils';
import {deleteNote} from '../api/bexNote';
import ActionsHeader from './ActionsHeader.vue';
import {notificationStore} from '../main'

vi.mock('../main', () => ({
    notificationStore: {
        $patch: vi.fn()
    }
}));

const directives = {
    tooltip: {}
};

const createWrapper = (props) => {
    return mount(ActionsHeader, {
        props,
        global: {
            stubs: {
                'prime-toolbar': true,
                'prime-button': true
            },
            directives
        }
    });
}

describe('ActionsHeader.vue', () => {
    const testNote = 'test-note';

    it('matches snapshot', () => {
        const wrapper = mount(ActionsHeader, {
            global: {
                stubs: {
                    'prime-toolbar': {
                        template: '<div class="prime-toolbar-stub"><slot name="start"></slot><slot name="end"></slot></div>'
                    },
                    'prime-button': true
                },
                directives
            }
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    describe('handleDeleteNote', () => {
        beforeEach(() => {
            vi.clearAllMocks();
        });

        it('calls deleteNote API and emits delete when selectedTitle matches title', async () => {
            vi.mock('../api/bexNote', () => ({
                deleteNote: vi.fn()
            }));
            const wrapper = createWrapper({
                selectedTitle: testNote,
                title: testNote,
                content: 'some content'
            });

            await wrapper.vm.handleDeleteNote();

            expect(deleteNote).toHaveBeenCalledWith(testNote);
            expect(wrapper.emitted('delete')).toBeTruthy();
            expect(wrapper.emitted('delete')).toHaveLength(1);
            expect(notificationStore.$patch).toHaveBeenCalledWith({
                type: 'warn',
                message: `"${testNote}" deleted successfully.`
            });
        });

        it('does not call deleteNote API when selectedTitle does not match title', async () => {
            const wrapper = createWrapper({
                selectedTitle: 'different-note',
                title: testNote,
                content: 'some content'
            });

            await wrapper.vm.handleDeleteNote();

            expect(deleteNote).not.toHaveBeenCalled();
            expect(wrapper.emitted('delete')).toBeFalsy();
            expect(notificationStore.$patch).not.toHaveBeenCalled();
        });
    });

    describe('handleSaveNote', () => {
        beforeEach(() => {
            vi.clearAllMocks();
        });

        it('calls saveNote API and emits save when selectedTitle is not set', async () => {
            vi.mock('../api/bexNote', () => ({
                saveNote: vi.fn(),
                deleteNote: vi.fn()
            }));
            const {saveNote, deleteNote} = await import('../api/bexNote');
            const wrapper = createWrapper({
                selectedTitle: '',
                title: testNote,
                content: 'new content'
            });

            await wrapper.vm.handleSaveNote();

            expect(saveNote).toHaveBeenCalledWith(testNote, 'new content');
            expect(deleteNote).not.toHaveBeenCalled();
            expect(wrapper.emitted('save')).toBeTruthy();
            expect(wrapper.emitted('save')).toHaveLength(1);
            expect(notificationStore.$patch).toHaveBeenCalledWith({
                type: 'success',
                message: `"${testNote}" saved successfully.`
            });
        });

        it('calls deleteNote and saveNote API and emits save when selectedTitle is set', async () => {
            vi.mock('../api/bexNote', () => ({
                saveNote: vi.fn(),
                deleteNote: vi.fn()
            }));
            const {saveNote, deleteNote} = await import('../api/bexNote');
            const wrapper = createWrapper({
                selectedTitle: 'old-note',
                title: testNote,
                content: 'updated content'
            });

            await wrapper.vm.handleSaveNote();

            expect(deleteNote).toHaveBeenCalledWith('old-note');
            expect(saveNote).toHaveBeenCalledWith(testNote, 'updated content');
            expect(wrapper.emitted('save')).toBeTruthy();
            expect(wrapper.emitted('save')).toHaveLength(1);
            expect(notificationStore.$patch).toHaveBeenCalledWith({
                type: 'success',
                message: `"${testNote}" saved successfully.`
            });
        });
    });

    describe('handleCreateNew', () => {
        it('emits createNew event when called', async () => {
            const wrapper = createWrapper({
                selectedTitle: '',
                title: '',
                content: ''
            });
            await wrapper.vm.handleCreateNew();
            expect(wrapper.emitted('createNew')).toBeTruthy();
            expect(wrapper.emitted('createNew')).toHaveLength(1);
        });
    });

});
