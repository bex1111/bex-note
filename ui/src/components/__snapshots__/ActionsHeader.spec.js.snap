// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionsHeader.vue > matches snapshot 1`] = `
"<div data-v-5adaa0a9=\\"\\" class=\\"prime-toolbar-stub\\"><span data-v-5adaa0a9=\\"\\" class=\\"header-title\\">Bex Note</span>
  <prime-button-stub data-v-5adaa0a9=\\"\\" icon=\\"pi pi-plus\\" severity=\\"secondary\\" class=\\"mr-2\\" text=\\"\\"></prime-button-stub>
  <prime-button-stub data-v-5adaa0a9=\\"\\" icon=\\"pi pi-save\\" severity=\\"secondary\\" class=\\"mr-2\\" text=\\"\\" disabled=\\"true\\"></prime-button-stub>
  <prime-button-stub data-v-5adaa0a9=\\"\\" icon=\\"pi pi-trash\\" severity=\\"danger\\" class=\\"mr-2\\" text=\\"\\" disabled=\\"true\\"></prime-button-stub>
</div>"
`;
